### DISCUSSION TITLE
WIQL Parent and Child Work Items

### DISCUSSION THREAD

**Discussion by stewarte19** [rfennell/AzurePipelines#1277] [Q&A]
Hello,

I am trying to get work items through their relationship to a parent item. [The Microsoft documentation](https://docs.microsoft.com/en-us/azure/devops/boards/queries/wiql-syntax?view=azure-devops) allows for the use of syntax to retrieve a source task, and its corresponding target (child) tasks. 

>([Source].[System.TeamProject] = 'MyProject')
>AND ([System.Links.LinkType] = 'System.LinkTypes.Dependency-Reverse')
>AND ([Target].[System.WorkItemType] <> ')

Here is an image of the query I'm using in the task.
![image](https://user-images.githubusercontent.com/78977273/161831699-a80ce7c3-3dff-4784-99dd-ea561b709428.png)

When I run this tasks it does not recognize [Source] as a valid field name.
![image](https://user-images.githubusercontent.com/78977273/161833008-34e0ee2d-60eb-411c-b1b5-951713603cde.png)

Could you please clarify how to use the query in this way?

Thank you.

**Comment by rfennell** [comment]
The problem is that this form of WIQL query 

```
SELECT [System.Id]
FROM WorkItemLinks
WHERE ([Source].[System.TeamProject] = 'MyProject')
AND ([System.Links.LinkType] = 'System.LinkTypes.Dependency-Reverse')
AND ([Target].[System.WorkItemType] <> '')
```

uses a special `FROM` section `FROM WorkItemLinks`.

The code in the task (and command line tester) builds the WIQL query using this form

`var wiqlQuery = `SELECT [System.Id] FROM workitems WHERE ${wiqlWhereClause} ORDER BY [System.ID] DESC`;`

so the `FROM` is incorrect.

There is no way without a code change to tasks to allow the `workitems' or `WorkItemLinks` to be set. 

I have added this to the backlog

**Comment by stewarte19** [comment]
Yes, this makes sense, hopefully it isn't too complex to implement. I'll keep an eye out for your product updates.

Thank you for the prompt reply.

**Reply by rfennell** [reply]
It is done and being deployed now, it will be version 3.82.5. There is now an extra parameter (a picklist) to choose the WIQL Target

**Reply by rfennell** [reply]
This is now live, the usage is

```
- task: richardfennellBM.BM-VSTS-XplatGenerateReleaseNotes.XplatGenerate-Release-Notes.XplatGenerateReleaseNotes@3
  inputs:
    wiqlWhereClause: '[Source].[System.WorkItemType] = ''Bug'''
    wiqlFromTarget: WorkItemLink
    ...
```

**Reply by stewarte19** [reply]
Ah, fantastic. I'll be implementing this sometime next week, great to have it done so quickly.
